src/core/ext/filters/stateful_session/stateful_session_filter.cc

```diff
@@ -158,29 +189,46 @@ ArenaPromise<ServerMetadataHandle> StatefulSessionFilter::MakeCallPromise(
               this, cookie_config->name->c_str(),
               std::string(*cookie_value).c_str());
     }
-    // We have a valid cookie, so add the call attribute to be used by the
-    // xds_override_host LB policy.
-    service_config_call_data->SetCallAttribute(
-        GetContext<Arena>()->New<XdsOverrideHostAttribute>(*cookie_value));
+    std::pair<absl::string_view, absl::string_view> host_cluster =
+        absl::StrSplit(*cookie_value, absl::MaxSplits(';', 1));
+    if (!host_cluster.first.empty() &&
+        OverrideClusterIfNeeded(host_cluster.second, service_config_call_data,
```

Comment:
I think we should set the host override independently of the cluster override.  Even if we're not setting the cluster override, it's conceivably possible that the same host could exist in whatever cluster we choose, so we should still set the host override.  And I think it makes the logic here a bit simpler to decouple the two.